sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3258,7 +3258,9 @@ class SQLConf extends Serializable with Logging {
   }
 
   def clear(): Unit = {
```

Comment:
This API is exposed to users, and make the SQLConf hard to restore to its initial state after calling `spark.sessionState.conf.clear`

Reply:
`SQLConf` is internnal.

Reply:
```scala
scala> spark.sessionState.conf.clear

scala> spark.sessionState.conf
res1: org.apache.spark.sql.internal.SQLConf = org.apache.spark.sql.internal.SQLConf@54e24511
```
show we make it private[sql] ?

Reply:
We don't. All classes under `org.apache.spark.sql.internal` are internal and users can still use. Spark marks things as internal by contract.

Reply:
OK.